go/vt/vtgate/plugin_mysql_server.go

```diff
@@ -649,6 +661,18 @@ func (srv *mysqlServer) shutdownMysqlProtocolAndDrain() {
 	}
 }
 
+// stopListener Close or Shutdown a mysql listener depending on the shutdown argument.
+func stopListener(listener *mysql.Listener, shutdown bool) {
+	if listener == nil {
+		return
+	}
+	if shutdown {
+		listener.Shutdown()
+	} else {
+		listener.Close()
+	}
```

Comment:
`listener.Shutdown()` calls `listener. Close()`. Any reason for doing this?

Reply:
`listener.Shutdown()` also set the `.shutdown` field to true and then close. We only want to close the listener here to stop accepting new connection but let the idle/current connection active. Shutting down will fail any idle connection on the next query.